grpc_server.go

```diff
@@ -196,16 +197,20 @@ func filterMetadata(hw []byte, filter string) ([]byte, error) {
 		if err, ok := v.(error); ok {
 			return nil, err
 		}
-		result = v
-	}
 
-	if resultString, ok := result.(string); ok { // if already a string, don't marshal
-		return []byte(resultString), nil
-	}
-	if result != nil { // if nil, don't marshal (json.Marshal(nil) returns "null")
-		return json.Marshal(result)
+		if vString, ok := v.(string); ok { // if already a string, don't marshal
+			result.WriteString(vString + "\n") // append v to result
+		} else if v != nil { // if nil, don't marshal (json.Marshal(nil) returns "null")
+			vByte, err := json.Marshal(v)
+			if err != nil {
+				return nil, err
+			}
+			result.Write(vByte) // append v to result
+			result.WriteRune('\n')
+		}
 	}
-	return nil, nil
+
+	return bytes.TrimSuffix(result.Bytes(), []byte("\n")), nil
```

Comment:
hmm how does this look to you instead?
```go
if v == nil {
    continue
}

switch vv := v.(type) {
case error:
    return nil, errors.Wrap(vv, ...)
case string:
    result.WriteString(vv)
default:
	vByte, err := json.Marshal(v)
	if err != nil {
		return nil, errors.Wrap(err, ...)
	}
	result.Write(vv)
}
result.WriteRune('\n')
```
I like it because it keeps all the type checking in one block. Either way the errors returned need to be `errors.Wrap`ped.

Reply:
this looks so much better honestly